test/test_argparse.py

```diff
@@ -56,6 +56,20 @@ def test_b(self):
             "function unsigned byte integer is less than minimum"
         )
 
+    def test_s(self):
+        import pytest
+        mod = self.make_parse_item("s", "const char*", "HPyUnicode_FromString")
+        assert mod.f("hello HPy") == "hello HPy"
+        with pytest.raises(ValueError) as err:
+            mod.f(b"hello\0HPy".decode('utf-8'))
+        assert str(err.value) == (
+            "function embedded null character"
+        )
+        with pytest.raises(TypeError) as err:
+            mod.f(b"hello HPy")
+        # No assertion of the message for now
+        # CPython's message reads: argument {N} must be str, not int
```

Comment:
This is a question of how far we want to go in compatibility of `HPyArg_Parse` with `PyArg_ParseTuple`. To have the same message we'd need to change the current structure of `argparse.c` a bit.

Reply:
I don't think that we need to have the same error message. It's important to have the same error type.

Reply:
I agree that the exact message is not important. Pinging @hodgestar since I think he's the author of the comment

Reply:
I am author of the comment :-) I think it may be useful to keep it just so we know. I intentionally did not make it a "TODO" comment, because I also think it's fine as it is.

Reply:
I don't think we need to have the same error message as PyArg_ParseTuple but I think we should check the error message here because this error message is generated by HPy code (i.e. our code) and we should check that the code is doing the right thing (i.e. generating a sensible error message with everything in the right place, etc).

Reply:
Fixed in the last commit